### ISSUE TITLE
ui: TPS on 10 min vs 1 hour is different

### ISSUE THREAD

**Issue by robert-s-lee** [cockroachdb/cockroach#28208]
**BUG REPORT**

Please describe the issue you observed, and any steps we can take to reproduce it:

Running YCBS Workload A.  10 minutes TPS vs 1 hours TPS are different

- Which version of CockroachDB are you using?

2.1 alpha

- What did you do?

look at 10min vs 1 hour TPS

- What did you expect to see?

the same TPS

- What did you see instead?

![screen shot 2018-08-02 at 12 56 53 pm](https://user-images.githubusercontent.com/6315124/43598848-bfc0ba64-9653-11e8-8b4f-bd2d67d4f859.png)
![screen shot 2018-08-02 at 12 56 42 pm](https://user-images.githubusercontent.com/6315124/43598849-bfce3644-9653-11e8-80ab-9841dbdaa425.png)

- What was the impact?

**Comment by knz** [comment]
cc @couchand

**Comment by couchand** [comment]
One thought that comes to mind immediately is the downsampling.  WIth a 1 hour history we'll have one sample per 30 seconds, with the 10 minute history we have a sample per 10 seconds, so if the load is spotty the extrema will get averaged out.

From the look of these charts that doesn't seem like enough to explain it, since the load looks to have been pretty constant for over a minute.

**Comment by couchand** [comment]
cc @mrtracy

**Comment by mrtracy** [comment]
Can we let it run a little longer and see if the 1 hour view gets up to 15k?

**Comment by mrtracy** [comment]
If we can confirm that the 1 hour graph reaches the correct number within the next two datapoints, I believe this is a visual artifact caused by using the AVG downsampler combined with taking derivatives. I think it could likely be mitigated by using a LAST downsampler - I can add support for that in a straightforward manner, it's already supported by the rollup system in anticipation of exposing it.

**Comment by robert-s-lee** [comment]
@piyush-singh can you work with @mrtracy and confirm.

**Comment by piyush-singh** [comment]
hi Robert, I'm trying to reproduce your issue above using roachprod, and I wanted to ask about your setup. I set up a simple 3 node cluster with a 4th node running ycsb workload A. I pointed ycsb to a haproxy loadbalancer in a simple roundrobin setup, but throughput isn't very high (500 reads/writes per second). The discrepancy between the 10 minute view and the 1 hour view is around 5%, but I suspect it might be larger for higher values, so I wanted to ask:

- What did your cluster configuration for the above look like? How many nodes, what types of machines, any flags you used, etc.
- What parameters did you set for ycsb (concurrency, splits, etc.)?

**Comment by piyush-singh** [comment]
Okay, Robert sent over a shell script to reproduce his exact setup yesterday. Taking a look, I was not able to reproduce the large discrepancy seen by Robert. I ran YCSB Workload A against an 18 node cluster that achieved 30k TPS throughput for 20 minutes. My "10 minutes" view of the SQL Queries chart corresponds very closely to the "1 hour" view:
﻿﻿﻿﻿﻿﻿﻿﻿﻿
<img width="871" alt="image" src="https://user-images.githubusercontent.com/21692034/43793755-77039a12-9a4a-11e8-8868-8c078f221dda.png">

<img width="864" alt="image" src="https://user-images.githubusercontent.com/21692034/43793808-9f06e10e-9a4a-11e8-972d-3e328319ed45.png">

Reducing the time the test was running to 4 minutes and then 2 minutes had no effect on the discrepancy between the two time selections (still within a small margin of error). I did notice the timestamp for the above ~20k throughput for reads and updates corresponds to the rightmost point on the bottom graph: is it possible the sampling point that would show the increased throughput hadn't been reached? Did the throughput at 16:35 in the "1 hour" view roughly correspond to throughput in the 10 minute view at the same timestamp?
